src/HasMedia/HasMediaCollectionsTrait.php

```diff
@@ -2,300 +2,48 @@
-//            $this->registerMediaCollections();
-//        }
-//
-//        collect($this->mediaCollections)->each(function (MediaCollection $mediaCollection) use ($media) {
-//            $actualMediaConversions = $this->mediaConversions;
-//
-//            $this->mediaConversions = [];
-//
-//            ($mediaCollection->mediaConversionRegistrations)($media);
-//
-//            $preparedMediaConversions = collect($this->mediaConversions)
-//                ->each(function (Conversion $conversion) use ($mediaCollection) {
-//                    $conversion->performOnCollections($mediaCollection->name);
-//                })
-//                ->values()
-//                ->toArray();
-//
-//            $this->mediaConversions = array_merge($actualMediaConversions, $preparedMediaConversions);
-//
-//        });
-//
-//        $this->registerMediaConversions($media);
-//
-//    }
-
-	/**
-	 * Register new Media Collection
-	 *
-	 * Adds new collection to model and set its name.
-	 *
-	 * @param $name
-	 *
-	 * @return MediaCollection
-	 * @throws MediaCollectionAlreadyDefined
-	 */
-	public function addMediaCollection( $name ): MediaCollection {
-
-//		if ($this->mediaCollections->has($name)) {
-//
-//		    throw new MediaCollectionAlreadyDefined("Error in $name");
-//		}
-
-//		V starej verzii to zbehlo 6 krat, teraz 12. Vola to aj FileAdder, vola to aj register Conversions,
-// pokazde sa to vola a preto asi nema zmysel chceckovat ci este nie su zaregistrovane. Nema to podla mna ani zmysel, vzdy sa len prepise kolekcia.
-
-		$collection = MediaCollection::create( $name );
-
-        $this->mediaCollections->put( $name, $collection );
-
-		return $collection;
-	}
-
-	/**
-	 * Returns a collection of Media Collections
-	 *
-	 * @return Collection|MediaCollection[]
-	 */
-	public function getMediaCollections(): Collection {
-		return $this->mediaCollections;
-	}
-
-	/**
-	 * Returns a Media Collection according to the name
-	 *
-	 * If Media Collection was not registered on this model, null is returned
-	 *
-	 * @param $name
-	 *
-	 * @return MediaCollection|null
-	 */
-	public function getMediaCollection( $name ): ?MediaCollection {
-
-		return $this->mediaCollections->get( $name );
-	}
+trait HasMediaCollectionsTrait
+{
+    use ParentHasMediaTrait;
+
+    /**
+     * This hooks the initialization to the right place
+     */
+    protected function bootIfNotBooted()
+    {
+        parent::bootIfNotBooted();
+
+        $this->initMediaCollections();
+    }
+
+
+    /**
+     * Initialize all collections for model
+     */
+    protected function initMediaCollections(): void
+    {
+        $this->mediaCollections = [];
+
+        $this->registerMediaCollections();
+    }
+
+    /**
+     * Register new Media Collection
+     *
+     * Adds new collection to model and set its name.
+     *
+     * @param $name
+     *
+     * @return MediaCollection
+     */
+    public function addMediaCollection($name): MediaCollection
```

Comment:
This is the place where in stead of spatie MediaCollecton object we use ours, so we also need to override this method.